core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1286,31 +1286,31 @@ class UnifiedLog(@volatile var logStartOffset: Long,
         // The first cached epoch usually corresponds to the log start offset, but we have to verify this since
         // it may not be true following a message format version bump as the epoch will not be available for
         // log entries written in the older format.
-        val earliestEpochEntry = leaderEpochCache.flatMap(_.earliestEntry)
+        val earliestEpochEntry = leaderEpochCache.flatMap(_.earliestEntry().asScala)
```

Comment:
To avoid multiple conversions, I think we can simply use Optional here and in the lines below?